shepherd_bms/src/compute.cpp

```diff
@@ -1,12 +1,48 @@
 #include "compute.h"
 
-ComputeInterface::ComputeInterface(){}
+ComputeInterface::ComputeInterface()
+{
+    initializeCAN(CANLINE_2, CHARGER_BAUD, &(this->chargerCallback));
+}
 
 ComputeInterface::~ComputeInterface(){}
 
-FaultStatus_t ComputeInterface::enableCharging(bool isEnabled){}
+void ComputeInterface::enableCharging(bool enableCharging){
 
-bool ComputeInterface::isCharging(){}
+    isChargingEnabled = enableCharging ? true : false;
+}
+
+
+FaultStatus_t ComputeInterface::sendChargingMessage(uint8_t voltageToSet, uint8_t currentToSet)
+{
+    if (!isChargingEnabled)
+    {
+        chargerMsg.cfg.chargerControl = CHARGE_DISABLED;
+        sendMessageCAN2(CANMSG_CHARGER, 8, chargerMsg.msg);
+        return isCharging() ? FAULTED : NOT_FAULTED; //return a fault if we DO detect a voltage after we stop charging
+    }
+
+    // equations taken from TSM2500 CAN protocol datasheet
+    chargerMsg.cfg.chargerVoltage = voltageToSet * 10;
```

Comment:
assuming this was checked already, but just making sure that this EQ results in the same Byte 2 and 3 that was produced in the earlier version